app/src/main/res/layout/fragment_game.xml

```diff
@@ -1,6 +1,127 @@
+            app:layout_constraintEnd_toEndOf="@id/tic_tac_toe_board"
+            app:layout_constraintHorizontal_bias="0.45"
+            app:layout_constraintStart_toStartOf="@id/tic_tac_toe_board"
+            app:layout_constraintTop_toTopOf="@id/tic_tac_toe_board"
+            app:layout_constraintVertical_bias="0.48000002" />
+
+        <GridLayout
+            android:id="@+id/tic_tac_toe_board"
+            android:rowCount="3"
+            android:columnCount="3"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/player_turn_view"
+            app:layout_constraintBottom_toTopOf="@id/button_end_game"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content" >
+
+            <ImageView
+                android:id="@+id/grid_upper_left"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/u_l_empty" />
+
+            <ImageView
+                android:id="@+id/grid_upper_center"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/u_c_empty" />
+
+            <ImageView
+                android:id="@+id/grid_upper_right"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/u_r_empty" />
+
+            <ImageView
+                android:id="@+id/grid_center_left"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/c_l_empty"/>
+
+            <ImageView
+                android:id="@+id/grid_center"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/c_empty" />
+
+            <ImageView
+                android:id="@+id/grid_center_right"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/c_r_empty" />
+
+            <ImageView
+                android:id="@+id/grid_lower_left"
+                android:layout_width="72dp"
+                android:layout_height="72dp"
+                android:layout_margin="22dp"
+                android:contentDescription="@string/l_l_empty" />
+
+            <ImageView
+                android:id="@+id/grid_lower_center"
+                android:layout_width="72dp"
```

Comment:
one thing you may want to use is a resource to qualify these image width / height. we are repeating '72dp' pretty often, so it'd be nice if in a dimens.xml file we could define a 'grid_height' dimension so we could only have that defined in one place

Reply:
Yes! I haven't gotten to making a dimen file or styles yet: I had that mentally tagged for the - Resolve accessibility issues/fine tune UI step that I laid out in my README but I didn't make that very clear. Is it better practice to polish the UI before working on the logic or vice versa?